validator/sawtooth_validator/journal/publisher.py

```diff
@@ -204,7 +207,22 @@ def finalize_block(self, identity_signing_key, pending_batches):
         committed_txn_cache = TransactionCache(self._block_store)
 
         builder = self._block_builder
+        bad_batches = []  # the list of batches that failed processing, used
         state_hash = None
+
+        # Walk the pending batch list:
+        # - find the state hash for the block, the block state_hash is
+        # is randomly placed on one of the transactions, so must interogate
+        # every batch to find it. If it is on a batch that failed processing
+        # then this block will be abandoned.
+        # - build three lists of batches.
+        # 1) a lists of all valid transactions that will be included in the
+        # block, these are added to the BlockBuilder to include in the Block
+        # 2) all batches that were not executed, these are to be returned
+        # in the pending_batches list
+        # 3) all batches that failed processing. These will be discarded. This
+        # list is needed in some case when the block is abandoned to make
+        # sure they do not remain in the pending_batches list.
```

Comment:
Indentation would help clarify this comment. It is currently very difficult to read.